api/app/src/routes/fhir-proxy.ts

```diff
@@ -19,51 +38,28 @@ const updateQueryString = (path: string, params: string): string | undefined =>
   return undefined;
 };
 
-// TODO make this dynamic/config/secret
-const router = proxy("http://fhir.staging.metriport.com", {
-  // const router = proxy("http://host.docker.internal:8888", {
-  proxyReqPathResolver: function (req) {
-    console.log(`ORIGINAL URL: `, req.url);
-    const parts = req.url.split("?");
-    const path = parts[0];
-    const queryString = parts.length > 1 ? updateQueryString(path, parts[1]) : undefined;
-    const updatedURL = "/fhir" + path + (queryString ? "?" + queryString : "");
-    console.log(`UPDATED URL: `, updatedURL);
-    return updatedURL;
-  },
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  userResDecorator: function (proxyRes, proxyResData, userReq, userRes) {
-    const payloadString = proxyResData.toString("utf8");
-    console.log(`ORIGINAL RESPONSE: `, JSON.stringify(JSON.parse(payloadString), undefined, 2));
-    // const orgRegex = /"Organization\/(.+)"/g;
-    // const orgReplace = `"Organization/urn:oid:$1"`;
-    // eslint-disable-next-line no-useless-escape
-    // const urlRegex = /https\:\/\/fhir\.staging\.metriport\.com/g;
-    // const urlReplace = `https://api.staging.metriport.com`;
-    const updatedPayload = payloadString;
-    // const updatedPayload = payloadString
-    //   // .replace(orgRegex, orgReplace)
-    //   .replace(urlRegex, urlReplace);
-    const payload = JSON.parse(updatedPayload);
-    if (payload.entry) {
-      // payload.entry
-      //   .filter((e: any) => e.resource?.resourceType === `DocumentReference`)
-      //   .forEach((e: any) => {
-      //     e.resource.id = URN_OID_PREFIX + e.resource.id;
-      //     // const contained = e.resource?.contained;
-      //     // if (contained) {
-      //     //   contained
-      //     //     .filter((c: any) => c.resourceType === `Organization`)
-      //     //     .forEach((c: any) => (c.id = URN_OID_PREFIX + c.id));
-      //     // }
-      //   });
-      // payload.entry
-      //   .filter((e: any) => e.resource?.resourceType === `Organization`)
-      //   .map((e: any) => e.resource)
-      //   .forEach((r: any) => (r.id = URN_OID_PREFIX + r.id));
-    }
-    console.log(`UPDATED RESPONSE: `, JSON.stringify(payload, undefined, 2));
-    return JSON.stringify(payload);
-  },
-});
+const router = fhirServerUrl
+  ? proxy(fhirServerUrl, {
+      proxyReqPathResolver: function (req) {
+        console.log(`ORIGINAL HEADERS: `, JSON.stringify(req.headers));
+        console.log(`ORIGINAL URL: `, req.url);
+        const parts = req.url.split("?");
+        const path = parts[0];
+        const queryString = parts.length > 1 ? updateQueryString(path, parts[1]) : undefined;
```

Comment:
should we check `parts.length === 2` ?

Reply:
I think we wouldn't want to fail if there are more than 2 parts... sure, this means the regex is flaky, so I assumed this would be safer.

Could improve on making `updteQueryString()` to work with all entries from position 1 onwards, but I think its not needed.

Cool?

Reply:
didn't realize that split returns an empty array with a single string if it couldn't be split

```
const emptyString = "";

// string is empty and separator is non-empty
console.log(emptyString.split("a"));
// [""]
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split

safe to ignore this comment